.github/workflows/plugins_react_tests.yml

```
@@ -13,48 +13,62 @@ permissions:
   contents: read
 
 jobs:
+  setup_matrix:
+    name: Setup matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.build_matrix.outputs.matrix }}
+    steps:
+      - name: Build test matrix
+        id: build_matrix
+        uses: ekohl/gha-matrix-builder@v0
+        with:
+          base_matrix_url: "https://raw.githubusercontent.com/${{ github.repository }}/${{ github.base_ref }}/.github/matrix.json"
```

I couldn't get it to run against my fork.
`github.repository` points at the target, so my branch name (`github.head_ref`) doesn't resolve here.
Maybe `github.sha` would work? But IMHO something we can fix later.

---

Weird, in my testing this worked:
```suggestion
          base_matrix_url: "https://raw.githubusercontent.com/${{ github.repository }}/${{ github.ref_name }}/.github/matrix.json"
```

But maybe I was only testing on push in my own repo so I didn't notice.

---

`ref_name` for this pr is `9881`, I tried.

---

https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables is a bit ambiguous. Perhaps this needs to be extracted (using non-trivial logic) from the PR. I'm more and more leaning to using `actions/checkout` by using https://github.com/actions/checkout#fetch-only-a-single-file

---

yes, this is also my preference, I think

---

I've opened https://github.com/ekohl/gha-matrix-builder/pull/2 to start trying this out.